boards/components/src/thread_network.rs

```diff
@@ -0,0 +1,255 @@
+// Licensed under the Apache License, Version 2.0 or the MIT License.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+// Copyright Tock Contributors 2022.
+
+//! Component to initialize the userland UDP driver.
+//!
+//! This provides one Component, UDPDriverComponent. This component initializes
+//! a userspace UDP driver that allows apps to use the UDP stack.
+//!
+//! Usage
+//! -----
+//! ```rust
+//!    let udp_driver = UDPDriverComponent::new(
+//!        board_kernel,
+//!        udp_send_mux,
+//!        udp_recv_mux,
+//!        udp_port_table,
+//!        local_ip_ifaces,
+//!        PAYLOAD_LEN,
+//!     )
+//!     .finalize(components::udp_driver_component_static!());
+//! ```
+
+// This buffer is used as an intermediate buffer for AES CCM encryption. An
+// upper bound on the required size is `3 * BLOCK_SIZE + radio::MAX_BUF_SIZE`.
+
+use capsules_core;
+use capsules_core::virtualizers::virtual_aes_ccm::MuxAES128CCM;
+use capsules_core::virtualizers::virtual_alarm::VirtualMuxAlarm;
+use capsules_extra::net::ipv6::ipv6_send::IP6SendStruct;
+use capsules_extra::net::network_capabilities::{
+    AddrRange, NetworkCapability, PortRange, UdpVisibilityCapability,
+};
+use kernel::hil::symmetric_encryption::{self, AES128Ctr, AES128, AES128CBC, AES128CCM, AES128ECB};
+
+use capsules_core::virtualizers::virtual_alarm::MuxAlarm;
+use capsules_extra::net::udp::udp_port_table::UdpPortManager;
+use capsules_extra::net::udp::udp_recv::MuxUdpReceiver;
+use capsules_extra::net::udp::udp_recv::UDPReceiver;
+use capsules_extra::net::udp::udp_send::{MuxUdpSender, UDPSendStruct, UDPSender};
+use core::mem::MaybeUninit;
+use kernel;
+use kernel::capabilities;
+use kernel::capabilities::NetworkCapabilityCreationCapability;
+use kernel::component::Component;
+use kernel::create_capability;
+use kernel::hil::radio;
+use kernel::hil::time::Alarm;
+
+const MAX_PAYLOAD_LEN: usize = super::udp_mux::MAX_PAYLOAD_LEN;
+pub const CRYPT_SIZE: usize = 3 * symmetric_encryption::AES128_BLOCK_SIZE + radio::MAX_BUF_SIZE;
+
+// Setup static space for the objects.
+#[macro_export]
+macro_rules! thread_network_driver_component_static {
```

Comment:
```suggestion
macro_rules! thread_network_component_static {
```

Or change the struct name.